quickstarts/microsoft.compute/vmss-windows-autoscale/main.bicep

```diff
@@ -0,0 +1,241 @@
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        addressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+        }
+      }
+    ]
+  }
+}
+
+resource publicIP 'Microsoft.Network/publicIPAddresses@2021-02-01' = {
+  name: publicIPAddressName
+  location: location
+  properties: {
+    publicIPAllocationMethod: 'Dynamic'
+    dnsSettings: {
+      domainNameLabel: longNamingInfix
+    }
+  }
+}
+
+resource loadBalancer 'Microsoft.Network/loadBalancers@2021-02-01' = {
+  name: loadBalancerName
+  location: location
+  properties: {
+    frontendIPConfigurations: [
+      {
+        name: 'LoadBalancerFrontEnd'
+        properties: {
+          publicIPAddress: {
+            id: publicIP.id
+          }
+        }
+      }
+    ]
+    backendAddressPools: [
+      {
+        name: bePoolName
+      }
+    ]
+    inboundNatPools: [
+      {
+        name: natPoolName
+        properties: {
+          frontendIPConfiguration: {
+            id: resourceId('Microsoft.Network/loadBalancers/frontendIPConfigurations', loadBalancerName, 'loadBalancerFrontEnd')
+          }
+          protocol: 'Tcp'
+          frontendPortRangeStart: natStartPort
+          frontendPortRangeEnd: natEndPort
+          backendPort: natBackendPort
+        }
+      }
+    ]
+  }
+}
+
+resource vmss 'Microsoft.Compute/virtualMachineScaleSets@2021-03-01' = {
+  name: vmssName
+  location: location
+  sku: {
+    name: vmSku
+    tier: 'Standard'
+    capacity: instanceCount
+  }
+  properties: {
+    overprovision: true
+    upgradePolicy: {
+      mode: 'Manual'
+    }
+    virtualMachineProfile: {
+      storageProfile: {
+        osDisk: {
+          createOption: 'FromImage'
+          caching: 'ReadWrite'
+        }
+        imageReference: imageReference
+      }
+      osProfile: {
+        computerNamePrefix: namingInfix
+        adminUsername: adminUsername
+        adminPassword: adminPassword
+      }
+      networkProfile: {
+        networkInterfaceConfigurations: [
+          {
+            name: nicName
+            properties: {
+              primary: true
+              ipConfigurations: [
+                {
+                  name: ipConfigName
+                  properties: {
+                    subnet: {
+                      id: resourceId('Microsoft.Network/virtualNetworks/subnets', virtualNetwork.name, subnetName)
+                    }
+                    loadBalancerBackendAddressPools: [
+                      {
+                        id: resourceId('Microsoft.Network/loadBalancers/backendAddressPools', loadBalancer.name, bePoolName)
+                      }
+                    ]
+                    loadBalancerInboundNatPools: [
+                      {
+                        id: '${loadBalancer.id}/inboundNatPools/${natPoolName}'
```

Comment:
use resourceId here as above

Reply:
inboundNatPools doesn't seem to be a resource, so I don't think I can use resourceId(), left as is

Reply:
Welcome to NRP - it is a resourceId - see: https://docs.microsoft.com/en-us/rest/api/compute/virtual-machine-scale-sets/create-or-update#virtualmachinescalesetipconfiguration or you can peek at the other samples in the repo